gcc/rust/rust-gcc.cc

```diff
@@ -1694,6 +1696,77 @@ Gcc_backend::array_constructor_expression (
   return ret;
 }
 
+// Build insns to create an array, initialize all elements of the array to
+// value, and return it
+tree
+Gcc_backend::array_initializer (tree fndecl, tree block, tree array_type,
```

Comment:
Can we make an issue to pull this out into rust-tree.h at some point it would be nice to eventually get rid of this abstraction since you have done most of the hard work for it :D